### PR TITLE
Fix CI + Use GitHub Actions to run the tests on Windows

### PR DESCRIPTION
**Pull request by aminya**
This uses GitHub Actions to install LLVM, set up the environment, and build and test Dpp. 

Here is an example of a successful run:
https://github.com/aminya/dpp/actions/runs/865871754

image

### CODE REVIEW

File: .github/workflows/d.yml [hunk 1]
```diff
@@ -3,44 +3,58 @@ on: [push, pull_request]
 
 jobs:
   test:
+    if: "!contains(github.event.head_commit.message, '[skip ci]')"
```
Comment (atilaneves) [inline .github/workflows/d.yml:6]: ?

File: .github/workflows/d.yml [hunk 2]
```diff
@@ -3,44 +3,58 @@ on: [push, pull_request]
...
-      - run: bundle install
+      - name: Setup Visual Studio Command Prompt - Windows
+        if: startsWith(matrix.os,'windows')
+        uses: ilammy/msvc-dev-cmd@v1
 
-      - name: Install D compiler
-        uses: dlang-community/setup-dlang@v1.0.5
+      # NOTE use KyleMayes/install-llvm-action once https://github.com/KyleMayes/install-llvm-action/pull/11 is merged
+      - name: Setup LLVM
+        uses: aminya/install-llvm-action@defeault-directory
```
Comment (atilaneves) [inline .github/workflows/d.yml:46]: The action itself is spelled incorrectly?

File: Gemfile.lock [hunk 3]
```diff
@@ -1,39 +0,0 @@
-GEM
```
Comment (atilaneves) [inline Gemfile.lock:1]: Why is this deleted?

File: appveyor.yml [hunk 4]
```diff
@@ -1,77 +1,5 @@
-platform: x64
```
Comment (atilaneves) [inline appveyor.yml:1]: It'd be easier to delete it.

File: .github/workflows/d.yml (hunk 2 above)
Comment (aminya) [reply .github/workflows/d.yml]: Oops 😅 Fixed in https://github.com/atilaneves/dpp/pull/287/commits/037ab3963471c5446e5c1dbccc0c05bf5c42fe76

File: appveyor.yml (hunk 4 above)
Comment (aminya) [reply appveyor.yml:1]: Appveyor will error if it cannot find appveyor.yml. If you prefer deleting the code, you should turn Appveyor off in the settings.
https://github.com/atilaneves/dpp/settings

File: .github/workflows/d.yml (hunk 1 above)
Comment (aminya) [reply .github/workflows/d.yml:6]: This allows using `[skip ci]` in the commit messages, which tells GitHub to skip running the CI.

File: Gemfile.lock (hunk 3 above)
Comment (aminya) [reply Gemfile.lock:1]: Because the lock file generated on Windows is not compatible with the lock file generated on Linux, and vice versa. If you want to use a fixed set of versions, it is better to fix the versions in Gemfile itself.

**Comment by aminya** [comment]
This last commit fixes the failing test on the master branch. It is the same one that I mentioned in 
https://github.com/atilaneves/dpp/pull/285#issuecomment-847292923

File: .github/workflows/d.yml [hunk 5]
```diff
@@ -43,7 +43,7 @@ jobs:
 
       # NOTE use KyleMayes/install-llvm-action once https://github.com/KyleMayes/install-llvm-action/pull/11 is merged
       - name: Setup LLVM
-        uses: aminya/install-llvm-action@defeault-directory
```
Comment (atilaneves) [inline .github/workflows/d.yml]: If this was mispelled, did it work before? Is it even needed?

**Review by atilaneves — APPROVED** [review]

File: .github/workflows/d.yml (hunk 5 above)
Comment (aminya) [reply .github/workflows/d.yml]: This is needed. The typo was actually in the branch name. I made a new branch with the correct name.
